pkg/server/grpc/server.go

```diff
@@ -92,14 +117,21 @@ func (s *Server) Run() error {
 // New creates a new instance of `IstioCAServiceServer`.
 func New(ca ca.CertificateAuthority, hostname string, port int) *Server {
 	return &Server{
-		ca:       ca,
-		hostname: hostname,
-		port:     port,
+		authenticator: &clientCertAuthenticator{},
+		authorizer:    &simpleAuthorizer{},
+		ca:            ca,
+		hostname:      hostname,
+		port:          port,
 	}
 }
 
 func (s *Server) createTLSServerOption() grpc.ServerOption {
+	cp := x509.NewCertPool()
+	cp.AppendCertsFromPEM(s.ca.GetRootCertificate())
+
 	config := &tls.Config{
+		ClientCAs:  cp,
+		ClientAuth: tls.VerifyClientCertIfGiven,
```

Comment:
I believe we should set `VerifyClientCertIfGiven` here instead of setting `AlwasyRequireClientCert`.

We have an authenticator on top of tls that actually makes use of the verified certificate. So for on prem, the authenticator will reject the request in case a client cert is missing.